### PR TITLE
:sparkle: add typescript definitions

### PR DESCRIPTION
**Pull request by AFaugeras**

### CODE REVIEW

File: typing/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+export declare type TypeDefinition = object;
```
Comment (fabienjuif) [inline typing/index.d.ts]: @AFaugeras someone does some typing for a lib that I wrote, and declared a module on top of all declarations : https://github.com/fabienjuif/use-bus/blob/master/index.d.ts

I think this is a must have so we don't have type collisions.
What is your though?

Comment (arthur-faugeras-cbp) [reply typing/index.d.ts]: @fabienjuif Sorry, didn't see you comment earlier 😕

I'm going to give it a try abd update my PRs subsequently 😉

File: package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "cmd:build": "microbundle --external all --sourcemap false -f es,cjs -i ",
     "build:core": "yarn cmd:build src/index.js",
     "build:helper": "yarn cmd:build src/helpers.js -o helpers/index.js",
-    "build": "npm-run-all --parallel build:*",
+    "build": "npm-run-all --parallel build:* && cp typing/index.d.ts dist/index.d.ts",
```
Comment (fabienjuif) [inline package.json]: This is better to add a "build:copy-types" I think:
1. This is less dependent of the OS/Shell
2. It works with npm-run-all (parallel execution here)

Comment (AFaugeras) [reply package.json]: Ok, will do ;)

Comment (AFaugeras) [reply package.json]: @fabienjuif do you want to support windows development on the library? If so, I will add a nodejs script to copy the file, if not I'll just leave it that way :)

Comment (fabienjuif) [reply package.json]: What do you think?

We can wait that someone struggle with it if you don't want to look into this.
If you have time, maybe fix it right away is a better solution.
As you wish @AFaugeras 👌

**Comment by AFaugeras** [comment]
Yes, that's what I was thinking: no need to publish it until all dependencies are ready to be published as well ;)

**Pull request merged**